core/imodel-bridge/src/imodelJsWrapper/ProductSettingsService.ts

```diff
@@ -0,0 +1,204 @@
+    );
+
+    if (result?.statusCode === 200) {
+      const documentMap: DocumentMapping = result.value;
+      if (documentMap !== null && documentMap !== undefined) {
+        const document = documentMap.properties.documents.find((doc) => doc.id.toLowerCase() === documentId.toLowerCase());
+
+        Logger.logInfo(
+          "iModelBridgeFwk Wrapper",
+          `[iModelBridgeService.NodeJsWrapper]: ProductSettingService:GetBriefCaseId--Document Map::${document}`
+        );
+        if (document !== null && document !== undefined) {
+          return document.briefcaseId;
+        }
+      }
+    }
+    return undefined;
+  }
+
+  private static async releaseAllIModelHubLocks(token: string, imodelId: string, briefCaseId: string): Promise<any> {
+    const url = URLResolver.imodelHubDeleteLockURL(imodelId, briefCaseId);
+    Logger.logInfo(
+      "iModelBridgeFwk Wrapper",
+      `[iModelBridgeService.NodeJsWrapper]: ProductSettingService:ReleaseAllIModelHubLocks-- ImodelHubDeleteLock Delete Request`
+    );
+    return HttpClientWrapper.deleteRequest(url, token);
+  }
+
+  private static async releaseSchemaLock(token: string, imodelId: string, briefCaseId: string): Promise<boolean | undefined> {
+    const url = URLResolver.imodelHubSchemaLockURL(imodelId, briefCaseId);
+
+    Logger.logInfo(
+      "iModelBridgeFwk Wrapper",
+      `[iModelBridgeService.NodeJsWrapper]: ProductSettingService:ReleaseSchemaLock-- ImodelHubSchema Lock Release URL ${url}`
+    );
+    return HttpClientWrapper.deleteRequest(url, token);
+  }
+}
+
+class HttpClientWrapper {
+
+  public static  async getRequest<T>(url: string, token: string, isTokenRequest = false): Promise<any> {
+    const res = req("GET", url, {
+      headers: { Authorization: token },
+    });
+
+    try {
+      const json = await res.getBody("utf8");
+      if (!isTokenRequest) {
+      }
+      if (json.statusCode === undefined) {
+        const jsonObj: T = JSON.parse(json);
+        return {statusCode: 200, value: jsonObj};
+      }
+      return {statusCode: json.statusCode, value: json.body};
+    } catch (err) {
+    }
+  }
+
+  public static  async deleteRequest(url: string, token: string): Promise<any>  {
+    try {
+      const res = req("DELETE", url, {
+        headers: { Authorization: token },
+      });
+      if (res.statusCode === 200)
+        return true;
+      return false;
+    } catch (err) {
+      return false;
+    }
+  }
+
+  public static  async postRequest(url: string, token: string, jsonObj: any): Promise<any>  {
+    const res = req("POST", url, {
+      headers: { Authorization: token },
+      json: jsonObj,
+    });
+    const json = await res.getBody("utf8");
+    return json.statusCode;
+  }
+}
+
+export class URLResolver {
+
+  public static productSettingsServiceDocumentURL(productId: string, contextId: string, iModel: string): string {
```

Comment:
We need to use the standard client for product settings service instead of a custom implementation

Reply:
Adam,
We need to remove the product settings briefcase Id management from the core. In the new design we are following, it is the responsibility of the caller to maintain this.